src/main/java/org/entur/lamassu/service/impl/GeoSearchServiceImpl.java

```diff
@@ -112,6 +137,25 @@ public List<Station> getStationsNearby(
     return stationCache.getAll(stationIds);
   }
 
+  @Override
+  public List<Station> getStationsInBoundingBox(
+    BoundingBoxQueryParameters boundingBoxQueryParameters,
+    StationFilterParameters stationFilterParameters
+  ) {
+    ReferencedEnvelope envelope = GeoUtils.mapToEnvelope(boundingBoxQueryParameters);
+    RangeQueryParameters rangeQueryParameters = GeoUtils.mapToRangeQueryParameters(
+      envelope
+    );
+    List<Station> stations = getStationsWithinRange(
+      rangeQueryParameters,
+      stationFilterParameters
+    );
+    return stations
+      .stream()
+      .filter(station -> envelope.contains(station.getLon(), station.getLat()))
```

Comment:
How does this handle boundaries?
Ideally, it would always be e.g. `>=` for two of the sides of the bbox, and strictly `<` for the other two, so that there are never overlaps or missing stations/vehicles.

Reply:
I suppose it is possible to simply filter out those with `latitude == maximumLatitude` and `longitude == maximumLongitude`

Reply:
As discussed in slack, envelope.contains includes all boundaries, so if the station/vehicle is *exactly* on the boundary it will be included. With the precision we're operating with such cases are very unlikely.